draft-ietf-tls-tls13.md

```diff
@@ -4128,9 +4212,174 @@ Section 7 of [RFC6066], as it is not applicable to AEAD ciphers and has
+to establish shared keys, the remainder of the handshake is encrypted,
+thus providing protection against passive attackers. Because the server
+authenticates before the client, the client can ensure that it only
+reveals its identity to an authenticated server. Note that implementations
+must use the provided record padding mechanism during the handshake
+to avoid leaking information about the identities due to length.
+
+The 0-RTT mode of operation generally provides the same security
+properties as 1-RTT data, with the two exceptions that the 0-RTT
+encryption keys do not provide full forward secrecy and that the
+the server is not able to guarantee full uniqueness of the handshake
+(non-replayability) without keeping potentially undue amounts of
+state. See {{early-data-indication}} for one mechanism to limit
+the exposure to replay.
+
+The reader should refer to the following references for analysis of the
+TLS handshake.
+
+## Record Layer {#security-record-layer}
+
+The record layer depends on the handshake producing a strong session
+key which can be used to derive bidirectional traffic keys and nonces.
+Assuming that is true, and the keys are used for no more data than
+indicated in [TODO] then the record layer should provide the following
+guarantees:
+
+Confidentiality.
+: An attacker should not be able to determine the plaintext contents
+of a given record. More formally it should not be able to distinguish
+between multiple encryptions of the same versus different plaintexts.
+(formally, IND-CPA security [REF]).
+
+Integrity.
+: An attacker should not be able to craft a new record which is
+different from an existing record which will be accepted by the receiver
+with more than negligible probability (formally, INT-CTXT security [REF]).
+
+Order protection/non-replayability
+: An attacker should not be able to cause the receiver to accept a
+record which it has already accepted or cause the receiver to accept
+record N+1 without having first processed record N.
+
+Length concealment.
+: Given a record with a given external length, the attacker should not be able
+to determine the amount of the record that is content versus padding.
+
+: Forward security after key change.
+If the traffic key update mechanism described in {{key-update}} has been
+used and the previous generation key is deleted, an attacker who compromises
+the endpoint should not be able to decrypt traffic encrypted with the old key.
+{:br}
 
-[[TODO]]
+Informally, TLS 1.3 provides these properties by AEAD-protecting the
+plaintext with a strong key. AEAD encryption {{RFC5116}} provides confidentiality
+and integrity for the data. Non-replayability is provided by using
+a separate nonce for each record, with the nonce being derived from
+the record sequence number ({{nonce}}), with the sequence
+number being maintained independently at both sides thus records which
+are delivered out of order result in AEAD deprotection failures.
```

Comment:
Is "deprotection" a standard term?  I get what you mean, but the word seems unusual.